aiidalab_widgets_base/viewers.py

```diff
@@ -167,15 +168,16 @@ class _StructureDataBaseViewer(ipw.VBox):
     selection = List(Int)
     selection_adv = Unicode()
     supercell = List(Int)
-    cell = Instance(Cell, allow_none=True)
+    cell = Instance(Cell, allow_none=True)  # Will be deprecated
```

Comment:
1. Do we need to add a deprecation warning then?
2. What is the motivation for this?

Reply:
per 1: Maybe (only if other app use this widget and directly use `cell` traitslet, @yakutovicha if the apps from empa using this directly). 

per 2: The motivation to replace `cell` with `ase_structure` is to read the symmetry information from the traitlets . From my point of view, it makes more sense to store and use full structure.

Reply:
I'll wait for @yakutovicha to comment on this.

Reply:
The `structure` trait of the `StructureDataViewer` child class serves the purpose of `ase_structure`. You can extract all the information from it.

> per 2: The motivation to replace `cell` with `ase_structure` is to read the symmetry information from the traitlets . From my point of view, it makes more sense to store and use full structure.

We actually thought of `_StructureDataBaseViewer` as a base class that could be used also for the trajectory viewers. I would, therefore, be very careful in changing this part.

I would at least do this particular change in another PR so that we could carefully consider all the pros and cons.

Reply:
Just to mention that @danielhollas is currently preparing things for the trajectory viewer, so we might want to revisit the design of `_StructureDataBaseViewer` class.

I can create a separate issue to collect the ideas. What do you think @unkcpz and @danielhollas?

Reply:
Thanks @yakutovicha please open an issue for trajectory. 
I will revert this part in the pr. Basically I need the ase.Atom only for parse the symmetry info.

Reply:
Look at the implementation of this PR again, if I revert back, there is no essential changes left.
I think if we call this class a 'StructureData' viewer, the positions info should be included with is not in the original implementation, which makes not much sense to me. 
@danielhollas Could you have a look at this implementation to see if this will conflict with your trajectory feature, if not, it is much easier to keep the base traitslet type change here. 
Therefore, I'll remove the `cell` traitslet if it is fine for you.

Reply:
> Look at the implementation of this PR again, if I revert back, there is no essential changes left.

Not really, I guess the purpose of this PR is to have the symmetry information displayed. This can stay.

>the positions info should be included with is not in the original implementation

sorry, I don't understand this, could you maybe rephrase?


> Just a correction to this, the structure trait can be either Atoms or AiiDA node, whereas `displayed_structure` is guaranteed to be the ASE Atoms object. So that's why I think it would make sense to move here.

No, it **_accepts_** both Atoms and AiiDA node, but then they get immediately converted into an ASE object:

https://github.com/aiidalab/aiidalab-widgets-base/blob/cabc5df932f41fcaf9907ac1ff2f0216ba20a700/aiidalab_widgets_base/viewers.py#L737-L738

> Regardless, I am fine with removing the cell traitlet. Since this is inside the private class, other things should not be using it anyway no?

It will also stay as a public member of the derived classes, we still need to be careful about removing it.

[P.S.]
Overall, I would like to stress that generally, I agree - we might change the classes to better satisfy our needs. We might do it in a separate PR (as I was suggesting) or just bite the bullet and do everything here. 

I try to be practical, that is why I suggest merging things that are clear and addressing the rest in a separate PR. If you want to do everything here - that is also fine with me. But keep in mind that it will cause some delays and long difficult-to-follow discussions.